beetsplug/importhistory.py

```diff
@@ -0,0 +1,157 @@
+                    item,
+                )
+            return
+        if (
+            not item["source_path"]
+            or item.mb_albumid in self.stop_suggestions_for_albums
+        ):
+            return
+        if os.path.isdir(item["source_path"]):
+            # We ask the user whether they'd like to delete the item's source
+            # directory
+            delete = input_yn(
+                "The item:\n{path}\nis originated in the directory:\n{source}\n"
+                "Would you like to delete the source directory of this item?".format(
+                    path=colorize_text(
+                        "text_warning", item.path.decode("utf-8")
+                    ),
+                    source=colorize_text(
+                        "text_warning", item["source_path"].decode("utf-8")
+                    ),
+                ),
+                require=True,
+            )
+            if delete:
+                self._log.info(
+                    "Deleting the item's source which is a directory: %s",
+                    item["source_path"],
+                )
+                rmtree(item["source_path"])
+        elif os.path.isfile(item["source_path"]):
+            # We ask the user whether they'd like to delete the item's source
+            # directory
+            print(
+                "The item:\n{path}\nis originated from:\n{source}\n"
+                "What would you like to do?".format(
+                    path=colorize_text(
+                        "text_warning", item.path.decode("utf-8")
+                    ),
+                    source=colorize_text(
+                        "text_warning", item["source_path"].decode("utf-8")
+                    ),
+                )
+            )
+            resp = input_options(
+                [
+                    "Delete the item's source",
+                    "Recursively delete the source's directory",
+                    "do Nothing",
+                    "do nothing and Stop suggesting to delete items from this album",
+                ],
+                require=True,
+            )
+            if resp == "d":
+                self._log.info(
+                    "Deleting the item's source file: %s", item["source_path"]
+                )
+                os.remove(item["source_path"])
+            elif resp == "r":
+                source_dir = os.path.dirname(item["source_path"])
+                self._log.info(
+                    "Searching for other items with a source_path attr containing: %s",
+                    source_dir,
+                )
+                # NOTE: I'm not sure why, but we need to escape it twice
+                # otherwise the search fails
+                escaped_source_dir = re.escape(
+                    re.escape(source_dir.decode("utf-8"))
+                )
+                source_dir_query = "source_path::^{}/*".format(
+                    escaped_source_dir
+                )
```

Comment:
It might be easier and a bit more reliable (and not require the mysterious double-escaping) to construct the query object directly. You should be able to do this:

```py
library.PathQuery("source_path", source_dir)
```

…and that should find all the items under `source_dir` according to the `source_path` field.

Reply:
Thanks for the suggestion, noted myself to test it as well.